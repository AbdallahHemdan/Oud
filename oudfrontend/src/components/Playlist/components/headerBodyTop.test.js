import React from 'react'
import HeaderBodyTop from './headerBodyTop';
import Enzyme, { shallow } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import checkPropTypes from 'check-prop-types'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new EnzymeAdapter() });


const setup = (props = {}) => {
    return shallow(<HeaderBodyTop {...props} />);
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
}
describe('HeaderBodyTop Component', () => {
    describe('testing HeaderBodyTop Component without props', () => {
        let component;
        beforeEach(() => {
            component = setup();
        })
        it("renders correctly without props", () => {
            const wrapper = findByTestAttr(component, "HeaderBodyTop");
            expect(wrapper.length).toBe(1);
        });
        it("renders title correctly without props", () => {
            const wrapper = findByTestAttr(component, "title");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe("");
        });

        it("renders owner correctly without props", () => {
            const wrapper = findByTestAttr(component, "owner");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe("");
        });
        it("renders credits correctly without props", () => {
            const wrapper = findByTestAttr(component, "credits");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe("By ")
        });
    });

    describe('testing HeaderBodyTop Component with props', () => {
        let component;
        let props = { title: "nice name", owner: "1" }
        beforeEach(() => {
            component = setup(props);
        })
        it("renders correctly with props", () => {
            const wrapper = findByTestAttr(component, "HeaderBodyTop");
            expect(wrapper.length).toBe(1);
        });
        it("renders title correctly with props", () => {
            const wrapper = findByTestAttr(component, "title");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe("nice name");
        });

        it("renders credits correctly with props", () => {
            const wrapper = findByTestAttr(component, "credits");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe("By ")
        });
    });
    describe('testing buttons', () => {
        let component;
        let props = { title: "nice name", owner: "1" }
        beforeEach(() => {
            component = setup(props);
            let wrapp = findByTestAttr(component, "owner");
            wrapp.simulate("click");
        })
        it("HeaderBodyTop shouldn't render", () => {
            const wrapper = findByTestAttr(component, "HeaderBodyTop");
            expect(wrapper.length).toBe(0);
        });
        it("title shouldn't render", () => {
            const wrapper = findByTestAttr(component, "title");
            expect(wrapper.length).toBe(0);
        });

        it("owner shouldn't render", () => {
            const wrapper = findByTestAttr(component, "owner");
            expect(wrapper.length).toBe(0);
        });
        it("credits shouldn't render", () => {
            const wrapper = findByTestAttr(component, "credits");
            expect(wrapper.length).toBe(0);
        });
    });
    describe('checking propTypes', () => {
        const propsT = { title: 'string', owner: '1' }
        const propsF = { title: {}, owner: 1 }
        it('should not throw a warning', () => {
            const result = checkPropTypes(HeaderBodyTop.propTypes, propsT, 'prop', HeaderBodyTop.name);
            expect(result).toBeUndefined();
        });
        it('should throw a warning', () => {
            const result = checkPropTypes(HeaderBodyTop.propTypes, propsF, 'prop', HeaderBodyTop.name);
            console.log(result);
            expect(result).toBeDefined();
        });
    });
    describe('snapshot test', () => {
        it('renders correctly', () => {
            const tree = renderer
                .create(<HeaderBodyTop />)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
