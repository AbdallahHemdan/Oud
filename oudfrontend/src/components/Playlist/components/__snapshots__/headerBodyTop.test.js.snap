// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBodyTop Component snapshot test renders correctly 1`] = `
<div
  className="playlistHeaderBodyTop"
  data-testid="HeaderBodyTop"
>
  <h2
    className="gray-text"
    data-testid="title"
  />
  <span
    className="whiteText"
    data-testid="credits"
  >
    By 
  </span>
  <button
    className="playlistAnchor songButton"
    data-testid="owner"
    onClick={[Function]}
  />
</div>
`;
